python/ray/_raylet.pyx

```diff
@@ -533,28 +541,28 @@ cdef execute_task(
             return execution_info.function(actor, *arguments, **kwarguments)
 
     return_ids = VectorToObjectIDs(c_return_ids)
-    with profiling.profile("task", extra_data=extra_data):
+    with core_worker.profile_event(b"task", extra_data=extra_data):
```

Comment:
If we're calling it like this here can we please do the same elsewhere and remove `profiling.profile` altogether?

Reply:
I also did it elsewhere in our codebase now, but we cannot remove `ray.profiling` since it is a public API that others could call to add stuff to the timeline.

Reply:
Note that doing the same elsewhere doesn't actually make it faster though. It only makes it faster if `core_worker` is typed as `CoreWorker`, in which case the python call overhead can be skipped.